src/sources/aws_ecs_metrics/mod.rs

```diff
@@ -0,0 +1,584 @@
+    let task = tokio::time::interval(Duration::from_secs(interval))
+        .take_until(shutdown)
+        .map(move |_| {
+            let client = Client::new();
+
+            let request = Request::get(&url)
+                .body(Body::empty())
+                .expect("error creating request");
+
+            let start = Instant::now();
+            let url2 = url.clone();
+            client
+                .request(request)
+                .and_then(|response| async move {
+                    let (header, body) = response.into_parts();
+                    let body = hyper::body::to_bytes(body).await?;
+                    Ok((header, body))
+                })
+                .into_stream()
+                .filter_map(move |response| {
+                    future::ready(match response {
+                        Ok((header, body)) if header.status == hyper::StatusCode::OK => {
+                            emit!(AwsEcsMetricsRequestCompleted {
+                                start,
+                                end: Instant::now()
+                            });
+
+                            let byte_size = body.len();
+                            let body = String::from_utf8_lossy(&body);
+
+                            match parser::parse(&body) {
+                                Ok(metrics) => {
+                                    emit!(AwsEcsMetricsReceived {
+                                        byte_size,
+                                        count: metrics.len(),
+                                    });
+                                    Some(stream::iter(metrics).map(Event::Metric).map(Ok))
+                                }
+                                Err(error) => {
+                                    emit!(AwsEcsMetricsParseError {
+                                        error,
+                                        url: url2.clone(),
+                                        body,
+                                    });
+                                    None
+                                }
+                            }
+                        }
+                        Ok((header, _)) => {
+                            emit!(AwsEcsMetricsErrorResponse {
+                                code: header.status,
+                                url: url2.clone(),
+                            });
+                            None
+                        }
+                        Err(error) => {
+                            emit!(AwsEcsMetricsHttpError {
+                                error,
+                                url: url2.clone(),
+                            });
+                            None
+                        }
+                    })
+                })
+                .flatten()
+        })
+        .flatten()
+        .forward(out)
+        .inspect(|_| info!("finished sending"));
+
+    Box::new(task.boxed().compat())
+}
+
+#[cfg(feature = "sinks-prometheus")]
```

Comment:
Given the parser tests, I think making these ones less specific are OK. It'd be nice to not depend on `sinks-prometheus` so the module can be built and tested independently.

You can see an example in the `apache_metrics` tests: https://github.com/timberio/vector/blob/45cb856d7645ec0918b4ab051d5138bd5842e8c1/src/sources/apache_metrics/mod.rs#L262-L480 which just test higher level conditions and rely on the parser tests for more detailed assertions.